doc/doxygen/src/Examples.dox

```diff
@@ -174,6 +174,57 @@
  * url = LocatableContentTrait(resolved_asset).getLocation()  # May be None
  * @endcode
  *
+ * @section examples_inspecting_a_reference Inspecting a reference
+ *
+ * This example shows how an entity reference of unknown provenance can
+ * be inspected to determine the qualities of the entity it points to.
+ *
+ * Let's say we're an audio application written in Qt, and someone has
+ * just dragged and dropped a line of text into the application. We want
+ * to know if the text is an entity reference, and if so, does it point
+ * to an audio clip.
+ *
+ * @code{.py}
+ * from openassetio.access import EntityTraitsAccess
+ * from openassetio_mediacreation.traits.content import LocatableContentTrait
+ * from openassetio_mediacreation.specifications.audio import SampledAudioResource
+ *
+ * class MyAudioApp(QMainWindow):
+ *
+ *   def __init__(self, manager):
+ *     super().__init__()
+ *     self.__manager = manager
+ *     self.__context = manager.createContext()
+ *     self.setAcceptDrops(True)
+ *     # [... Other Qt setup]
+ *
+ *   def dragEnterEvent(self, event):
+ *    if event.mimeData().hasFormat("text/plain"):
```

Comment:
[consider] `text/uri-list` seems to [be a thing](https://www.iana.org/assignments/media-types/text/uri-list).